fetch/cache.go

```diff
@@ -70,6 +70,9 @@ func (hpc *HashPeersCache) Get(hash types.Hash32) (HashPeers, bool) {
 
 // GetRandom returns random peer for a given hash.
 func (hpc *HashPeersCache) GetRandom(hash types.Hash32, rng *rand.Rand) (p2p.Peer, bool) {
+	hpc.mu.Lock()
+	defer hpc.mu.Unlock()
+
 	hashPeersMap, exists := hpc.Get(hash)
```

Comment:
it's bad design to have a public API Get that's not thread-safe, but others are (GetRandom/Add).
i think you should rename `Get` to `getWithStats` instead.

Reply:
now we have get, getWithStats and Get as well as add and Add (the former needed in AddPeersFromHash)

Reply:
hmmmmm... no. my comment above is to `rename`... 

we should only have
GetRandom/RegisterPeerHashes/AddPeersFromHash as the public API of cache.go. they all take a lock at the beginning.
then there is get/getWithStats/add that do not take locks.

Reply:
or consider adding a boolean to `get()` to indicate whether to update stats or not.

Reply:
Removed cache.Get, added getCachedEntry helper in tests as you suggested :) cheers. Plus added tests for GetRandom/RegisterPeerHashes/AddPeersFromHash.